src/storage-client/src/source/persist_source.rs

```diff
@@ -266,6 +270,12 @@ impl PendingWork {
                             |time| !until.less_equal(time),
                             row_builder,
                         ) {
+                            if is_filter_pushdown_audit {
+                                // Ideally we'd be able to include the part stats here, but that
+                                // would require us to exchange them around. It's unclear if that's
+                                // worth it for work that's already known to be unnecessary.
```

Comment:
Threading around / logging the `PartialBatchKey` might be enough to be useful, but I agree this is fine the way it is.

Reply:
that's a great idea! no guarantee that the part sticks around long enough for us to look at it, but definitely better than nothing